static-main/js/search.js

```diff
@@ -0,0 +1,166 @@
+import Fuse from 'fuse.js';
+
+import $ from './helpers/jq-helpers';
+
+class SynaSearch {
+  constructor({ queryParam, searchInput, resultsContainer, template }) {
+    this.searchInput = $(searchInput);
+    this.resultsContainer = $(resultsContainer);
+    this.template = $(template);
+    this.fuseOptions = {
+      shouldSort: true,
+      matchAllTokens: true,
+      includeMatches: true,
+      tokenize: true,
+      threshold: 0.2,
+      location: 0,
+      distance: 100,
+      maxPatternLength: 32,
+      minMatchCharLength: 4,
+      keys: [
+        { name: 'title', weight: 0.8 },
+        { name: 'contents', weight: 0.5 },
+        { name: 'tags', weight: 0.3 },
+        { name: 'categories', weight: 0.3 }
+      ]
+    };
+  
+    this.summaryInclude = 60;
+    this.indexCache = null;
+
+    const searchQuery = this.param(queryParam) || '';
+    this.searchInput.val(searchQuery.trim());
+    this.searchInput.on('input', e => this.search(e.target.value.trim()));
+    this.search(searchQuery);
+  }
+
+  getIndex(callback) {
+    if (this.indexCache) {
+      return callback(this.indexCache);
+    }
+
+    $.ajax({ method: 'get', url: '/index.json' }).then(data => {
+      this.indexCache = data;
+      callback(data);
+    });
+  }
+
+  search(query) {
+    if (!query) {
+      return this.resultsContainer.html('<p>Please enter a word or phrase above</p>');
```

Comment:
Probably hard to make that language agnostic later on. Aka we can't just use `i18n`.